src/OpenSage.Game/Data/Map/AssetPropertyCollection.cs

```diff
@@ -20,12 +19,28 @@ internal static AssetPropertyCollection Parse(BinaryReader reader, MapParseConte
             return new AssetPropertyCollection(result);
         }
 
-        public AssetProperty this[string name] => this.FirstOrDefault(x => x.Key.Name == name);
-
         internal AssetPropertyCollection(IList<AssetProperty> list)
-            : base(list)
         {
+            foreach (var property in list)
+            {
+                Add(property);
+            }
+        }
 
+        protected override string GetKeyForItem(AssetProperty item)
+        {
+            return item.Key.Name;
+        }
+
+        // TODO: Remove this if we retarget the library to .NET Core 2.0
+        public bool TryGetValue(string key, out AssetProperty value)
+        {
+            return Dictionary.TryGetValue(key, out value);
+        }
+
+        public AssetProperty GetPropOrNull(string key)
```

Comment:
Doesn't matter that much, but in general, I prefer full names for things, i.e. `GetPropertyOrNull`.